scRNA-seq/04-scrnaseq_exercise.Rmd

```diff
@@ -0,0 +1,342 @@
+---
+title: "Exercise: Gene markers in single-cell data"
+output:   
+  html_notebook: 
+    toc: true
+    toc_float: true
+---
+
+Our objective for this exercise is to investigate the expression of marker genes
+across cell types.
+Here we consider *marker genes* to be genes considered by the field to be 
+classic indicators of cell type.
+
+These *marker genes* are the ones often used for FAC sorting, which is what
+was done in [Darmanis et al. _Cell Reports._ 2017.](https://www.ncbi.nlm.nih.gov/pubmed/29091775)
+We previously normalized and filtered these data in `01-normalizing_scRNA`. 
+Here's a figure from their paper to give you a very brief overview of their 
+FAC sorting methodology. 
+  
+![**Figure 1B from Darmanis et al. _Cell Reports._ 2017**](data/gbm_figure.jpg)
+
+For more details on the methods, [see their manuscript.](https://www.ncbi.nlm.nih.gov/pubmed/29091775)
+But, in short, this study splits cells from a glioblastoma tumor by their 
+cell-type based on their FAC sorting. 
+
+The authors identified neoplastic cells using multiple analyses, including a
+high level of _SOX9_ expression. 
+The neoplastic cells make up the vast majority of the tumor core. 
+So by seeing what marker genes they express, we may get an initial, gene-by-gene
+idea of how these cells behave. 
+
+After extracting the normalized data for our marker genes of interest, we will 
+make a heatmap to compare the mean expression of the marker genes by cell type.
+Then we will visualize the marker gene expression in reduced dimensions by 
+creating a principal components scatterplot that color codes each cell by the 
+expression level of a marker gene. 
+
+## Setup 
+
+We'll be using the `%>%` operator (pipe) and we will need to set the seed so our 
+results are reproducible.
+
+```{r}
+# Set the seed
+set.seed(1234)
+
+# Load the pipe
+`%>%` <- dplyr::`%>%`
+```
+
+Use the template code chunk below to make a new directory where you can save plots.
+
+```{r eval=FALSE}
+plots_dir <- <DIRECTORY_NAME>
+if (!dir.exists(<FILL_IN_THE_BLANK>)) {
+  dir.create(<FILL_IN_THE_BLANK>)
+}
+```
+
+## Import and set up the data and metadata
+
+Use `readr`'s function for reading RDS files. 
+Import the `SingleCellExperiment` object that we stored in "sce_norm.RDS" in the
+`data` folder in `01-normalizing_scRNA-seq`. 
+Remember to use `file.path` and `::` to call any packages that you might use that
+you haven't imported.
+
+```{r eval=FALSE}
```

Comment:
If participants go to run all, won't the inclusion of `eval=FALSE` hamper them?

Reply:
Yes. It will. I will delete all those.

Reply:
Okay, was checking. I use it less often than you do.